File: Programmers/다음큰숫자.py
def binary(n):
    n_binary = ''
    while n != 1:
        n_binary = str(n % 2) + n_binary
        n = n // 2
    return n_binary.count('1') + 1

def solution(n):
    
    n_count_one = binary(n)
    answer = n + 1
    while True:
        answer_count_one = binary(answer)
        if answer_count_one == n_count_one:
            return answer
        answer += 1

File: JavaScript/다음큰숫자.js
const binary = (num) => {
  let result = "";
  while (num !== 1) {
    result += num % 2;
    num = Math.floor(num / 2);
  }
  return "1" + result;
};

function solution(n) {
  const count = binary(n)
    .split("")
    .filter((s) => {
      return s === "1";
    }).length;
  while (true) {
    n += 1;
    if (
      count ===
      binary(n)
        .split("")
        .filter((s) => {
          return s === "1";
        }).length
    ) {
      return n;
    }
  }
}

console.log(solution(78));
